asapdiscovery-dataviz/asapdiscovery/dataviz/html_viz.py

```diff
@@ -278,20 +279,34 @@ def get_interactions_plip(self, pose) -> dict:
         because it's v sensitive to protonation. PLIP does protonation itself.
         """
         # create the complex PDB file.
-        save_openeye_pdb(combine_protein_ligand(self.protein, pose), "tmp_complex.pdb")
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            tmp_pdb = os.path.join(tmpdirname, "tmp_complex.pdb")
+            save_openeye_pdb(combine_protein_ligand(self.protein, pose), tmp_pdb)
 
-        # run the PLIP CLI.
-        subprocess.run(["plip", "-f", "tmp_complex.pdb", "-x", "-o", "tmp"])
+            # run the PLIP CLI.
+            subprocess.run(["plip", "-f", tmp_pdb, "-x", "-o", tmpdirname])
 
-        # load the XML produced by PLIP that contains all the interaction data.
-        intn_dict_xml = xmltodict.parse(
-            ET.tostring(ET.parse("tmp/report.xml").getroot())
-        )
+            # load the XML produced by PLIP that contains all the interaction data.
+            intn_dict_xml = xmltodict.parse(
+                ET.tostring(ET.parse(os.path.join(tmpdirname, "report.xml")).getroot())
+            )
 
         intn_dict = {}
         intn_counter = 0
         # wrangle all interactions into a dict that can be read directly by 3DMol.
-        for _, data in intn_dict_xml["report"]["bindingsite"]["interactions"].items():
+        binding_site_data = intn_dict_xml["report"]["bindingsite"]
+        if isinstance(binding_site_data, list):
+            # this can happen if multiple binding sites (e.g. dimer) exist
+            interaction_data = {}
+            [
+                interaction_data.update(binding_site["interactions"])
+                for binding_site in binding_site_data
+            ]
```

Comment:
Is the intention here to just overwrite the dict with the values of the last `'interactions'` dict in `binding_site`? Is this guaranteed to always be what we want to do?

Reply:
keys here are unique, we're enumerating interactions when we get them from PLIP. so in this way, no interaction will be overwritten at any point.